### DISCUSSION TITLE
How to CREATE TEMPORARY TABLE?

### DISCUSSION THREAD

**Discussion by jrmidkiff** [sqlalchemy/sqlalchemy#10554] [Usage Questions] [answered]
The various discussions and google searching have not turned up a clear answer for me - how can I use either sqlalchemy core or ORM to create a temporary table according to the SQL standard? I don't want to have to ever have to literally type "CREATE TEMPORARY TABLE"

The peewee ORM has an easy argument to pass `temporary=True` when [creating a table](https://github.com/coleifer/peewee/issues/1404), but I can't find anything similar for SQLAlchemy

I generally use postgres and oracle

**Comment by gordthompson** [comment]
I've seen people use

```python
temp_tbl = Table(
    "my_temp_table",
    MetaData(),
    Column("id", Integer, … ),
    … ,
    prefixes=["TEMPORARY"],
)
temp_tbl.create(engine)
```

**Reply by jrmidkiff** [reply]
Sure, you could always prefix them - but that still means the table won't be a _true_ temporary table that is only visible for the connection and is auto-deleted

**Reply by CaselIT** [reply, marked as answer]
>  but that still means the table won't be a true temporary table that is only visible for the connection and is auto-deleted

what do you mean? the above executes `create temporary table ...` so it's a temp table from the db point of view.

**Reply by jrmidkiff** [reply]
Ohhhhh, I _totally_ misunderstood what `prefixes` does. Thank you!
